scripts/screens/ChangeGenderScreen.py

```diff
@@ -0,0 +1,631 @@
+                    self.selected_cat_elements["cat_gender"] = pygame_gui.elements.UITextBox(
+                        f"{self.the_cat.genderalign}",
+                        scale(pygame.Rect((252, 500), (500, 500))),
+                        object_id=get_text_box_theme("#text_box_30_horizcenter_spacing_95"),
+                        manager=MANAGER
+                    )
+
+            elif event.ui_element == self.buttons["add_pronouns"]:
+                PronounCreation(self.the_cat)
+                self.previous_cat_button.disable()
+                self.next_cat_button.disable()
+                self.back_button.disable()
+
+            else:
+                with open('resources/dicts/pronouns.json', 'r', encoding='utf-8') as file:
+                    pronouns_data = ujson.load(file)
+
+                for pronounset in pronouns_data["default_pronouns"]:
+                    add_button_id = f"add_button_{pronounset['ID']}" + "preset"
+                    if event.ui_element == self.buttons.get(add_button_id):
+                        print("added")
+                        for x in range(len(pronouns_data["default_pronouns"])):
+                            if pronounset['ID'] == pronouns_data["default_pronouns"][x]['ID']:
+                                print("in default")
+                                if not self.is_duplicate(pronounset):
+                                    self.the_cat.pronouns.append(pronounset)
+                                    self.update_selected_cat()
+                for pronounset in game.clan.custom_pronouns:
+                    remove_button_id = f"remove_button_{pronounset['ID']}" + "preset"
+                    add_button_id = f"add_button_{pronounset['ID']}" + "preset"
+                    if event.ui_element == self.buttons.get(add_button_id):
+                        print("added")
+                        for x in range(len(game.clan.custom_pronouns)):
+                            if pronounset['ID'] == game.clan.custom_pronouns[x]['ID']:
+                                print("in custom")
+                                if not self.is_duplicate(pronounset):
+                                    self.the_cat.pronouns.append(pronounset)
+                                    self.update_selected_cat()
+                    elif event.ui_element == self.removalbuttons.get(remove_button_id):
+                        game.clan.custom_pronouns.remove(pronounset)
+                        self.update_selected_cat()
+                for pronounset in self.the_cat.pronouns:
+                    remove_button_id = f"remove_button_{pronounset['ID']}"
+                    if event.ui_element == self.removalbuttons.get(remove_button_id):
+                        self.the_cat.pronouns.remove(pronounset)
+                        self.update_selected_cat()
+                        break
+
+        elif event.type == pygame.KEYDOWN and game.settings['keybinds']:
```

Comment:
Probably best to remove keybinds for now. They are a pain and a half. As this code is currently set-up, it would allow game.switches['cat'] to be set to an invalid cat ID.

Reply:
Removed :D